pkg/daemon/daemon.go

```diff
@@ -256,7 +259,16 @@ func (d *daemon) processNetworkGUID(networkID string, spec *utils.IbSriovCniSpec
 	} else {
 		guidAddr, err = d.guidPool.GenerateGUID()
 		if err != nil {
-			return fmt.Errorf("failed to generate GUID for pod ID %s, with error: %v", pi.pod.UID, err)
+			switch err {
+			// If the guid pool is exhausted, need to sync with SM in case there are unsynced changes
+			case guid.GuidPoolExhaustedError:
+				err = syncGuidPool(d.smClient, d.guidPool)
```

Comment:
is there a chance we end up in an in-consistent state here ?

im thinking in case DeletePeriodicUpdate is called at the same time.

Reply:
I'd say it doesn't really matter. `DeletePeriodicUpdate` doesn't introduce any new guids to the pool which could be deleted during a call to `Reset` and overwritten when a GUID for the next pod needs to be generated. Moreover, it makes an API call to UFM to release the GUID regardless of whether it is in the pool or not.
The model introduced with this fix offers eventual consistency for the GUID pool - so those changes would be reflect when the next sync occurs.
WDYT?

Reply:
ack so worst we will have some GUIDs in the pool that are not in use but will be marked as allocated.